src/Control/Middleware/HTTPCacheControlMiddleware.php

```diff
@@ -0,0 +1,765 @@
+    ];
+
+    /**
+     * Default forcing level
+     *
+     * @config
+     * @var int
+     */
+    private static $defaultForcingLevel = 0;
+
+    /**
+     * Forcing level forced, optionally combined with one of the below.
+     */
+    const LEVEL_FORCED = 10;
+
+    /**
+     * Forcing level caching disabled. Overrides public/private.
+     */
+    const LEVEL_DISABLED = 3;
+
+    /**
+     * Forcing level private-cached. Overrides public.
+     */
+    const LEVEL_PRIVATE = 2;
+
+    /**
+     * Forcing level public cached. Lowest priority.
+     */
+    const LEVEL_PUBLIC = 1;
+
+    /**
+     * Forcing level caching enabled.
+     */
+    const LEVEL_ENABLED = 0;
+
+    /**
+     * A list of allowed cache directives for HTTPResponses
+     *
+     * This doesn't include any experimental directives,
+     * use the config system to add to these if you want to enable them
+     *
+     * @config
+     * @var array
+     */
+    private static $allowed_directives = [
+        'public',
+        'private',
+        'no-cache',
+        'max-age',
+        's-maxage',
+        'must-revalidate',
+        'proxy-revalidate',
+        'no-store',
+        'no-transform',
+    ];
+
+    /**
+     * Get current vary keys
+     *
+     * @return array
+     */
+    public function getVary()
+    {
+        // Explicitly set vary
+        if (isset($this->vary)) {
+            return $this->vary;
+        }
+
+        // Load default from config
+        $defaultVary = $this->config()->get('defaultVary');
+        return array_keys(array_filter($defaultVary));
+    }
+
+    /**
+     * Add a vary
+     *
+     * @param string|array $vary
+     * @return $this
+     */
+    public function addVary($vary)
+    {
+        $combied = $this->combineVary($this->getVary(), $vary);
+        $this->setVary($combied);
+        return $this;
+    }
+
+    /**
+     * Set vary
+     *
+     * @param array|string $vary
+     * @return $this
+     */
+    public function setVary($vary)
+    {
+        $this->vary = $this->combineVary($vary);
+        return $this;
+    }
+
+    /**
+     * Combine vary strings/arrays into a single array, or normalise a single vary
+     *
+     * @param string|array[] $varies Each vary as a separate arg
+     * @return array
+     */
+    protected function combineVary(...$varies)
+    {
+        $merged = [];
+        foreach ($varies as $vary) {
+            if ($vary && is_string($vary)) {
+                $vary = array_filter(preg_split("/\s*,\s*/", trim($vary)));
+            }
+            if ($vary && is_array($vary)) {
+                $merged = array_merge($merged, $vary);
+            }
+        }
+        return array_unique($merged);
+    }
+
+
+    /**
+     * Register a modification date. Used to calculate the Modification-Date http header
+     *
+     * @param string|int $date Date string or timestamp
+     * @return HTTPCacheControlMiddleware
+     */
+    public function registerModificationDate($date)
```

Comment:
What value does this provide over `addHeader('Last-Modified")`?

Reply:
It lets you call it multiple times and does the calculation to get only the latest modification time.

Reply:
Right, so it saves you a conditional in custom code. I guess its obvious enough to developers. I've clarified the PHPDoc, it talked about the "Modification-Date http header"